compute_endpoint/globus_compute_endpoint/endpoint/endpoint_manager.py

```diff
@@ -115,6 +162,23 @@ def __init__(
             )
             exit(os.EX_DATAERR)
 
+        self._mt_user = pwd.getpwuid(os.getuid())
+        if config.force_mt_allow_same_user:
+            self._allow_same_user = True
+            log.warning(
+                "Configuration item `force_mt_allow_same_user` set to true; this is"
+                " considered a very dangerous override -- please use with care,"
+                " especially if allowing this endpoint to be utilized by multiple"
+                " users."
+                f"\n  Endpoint (UID, GID): ({os.getuid()}, {os.getgid()}) "
+            )
+        else:
```

Comment:
This is an overall 'Am I a privileged user' logic.  Instead of buried in an 'else' block in class init, IMO it's cleaner to centralize this in its own method/class.

That would also make it easier to invoke/test etc.

Reply:
Good point; thanks for that.  Done.

Reply:
FWIW, the original proposed change was just going to be the flag in the configuration with the main purpose of reducing log noise.  But after discussion with another dev, I amended that commit and ... arrived at this PR.  So, thanks for the observation.

I've also updated the PR description.